omaha/goopdate/download_manager.cc

```diff
@@ -530,7 +533,17 @@ HRESULT DownloadManager::CachePackage(const Package* package,
   const CString package_name(package->filename());
   PackageCache::Key key(app_id, version, package_name);
 
-  HRESULT hr = package_cache()->Put(
+  HRESULT hr;
+  if (ConfigManager::Instance()->AlwaysVerifyAuthenticodeSignatures()) {
+    hr = EnsureSignatureIsValid(*source_file_path);
+    if (FAILED(hr)) {
+      CORE_LOG(LE, (_T("[EnsureSignatureIsValid failed][%s][0x%08x]"),
+                    package->filename(), hr));
+      return SIGS_E_INVALID_SIGNATURE;
```

Comment:
Perhaps we should define a new error to disambiguate between hash and Authenticode failures.

Reply:
My preference would be to forego this complexity for now. But if it's necessary for you to merge then I'll look into it.

Reply:
Yes, we need to define a new value to return in this case, since we want to know if it failed the hash or Authenticode. It's fairly easy to create a new symbol in base/error.h, grouped by the Download Manager errors.

Reply:
Resolved in ae82ccc708f85befa85a3c2f6c9e71f64c760c40.